src/EntwineLLM/Servers/LlmServerBase.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace EntwineLlm.Models
+{
+    public abstract class LLMServerBase(string name, Uri baseUri)
+    {
+        public string Name { get; } = name;
+        public Uri BaseUrl { get; set; } = baseUri;
+        public TimeSpan RequestTimeOut { get; set; } = new TimeSpan(0, 10, 0);
+
+        public abstract Task<string[]> GetModelListAsync();
```

Comment:
At the current state, you fully right. But some server have spécifique implementation some time. I thinks about that another serve I wish add is Petals. But petals have not realy "chat api". It use Text Generation instead.
In real, I thin that more code need to move to LLMServerBase class. But curently I'm not confident of what to externalise.

Reply:
[chat.petals.dev](https://chat.petals.dev/)

Reply:
Will look at it asap, thank you

Reply:
> At the current state, you fully right. But some server have spécifique implementation some time. I thinks about that another serve I wish add is Petals. But petals have not realy "chat api". It use Text Generation instead. In real, I thin that more code need to move to LLMServerBase class. But curently I'm not confident of what to externalise.

Didn't know about Petals, looking at it i saw it uses a distributed LLM among users. While extremely useful perfomance-wise, i'm a little concerned about sending prompts and receiving requests from external sources, which is the reason why i wrote EntwineLLM . Let's think about the real need to include Petals

Reply:
Petals use HiveMind to distibute an LLM over multiple node. It's real that if one use the public swarm, the data is also public. To avoid this user can configure a private swarm. I personaly use private swarm  of Petals that connect 6 of my GPU.

Here, I wish to have more support over LLM server. An most server has some OpenAPI compatible end points and sometime other end point more powerfull. I think that have an ancestor class for LLM server inherited for specific implementation is not a bad way. Except if every server use the same API...

Reply:
Ok, thank you. I think we can resolve this comment, let's keep the implementation of GetModelListAsync method on each class